### PR TITLE
Initialize gatsby site

### PR DESCRIPTION
**Pull request by jonathansick**
The purpose of this PR is to set up the Gatsby site and re-create the basic structure of the current learn.astropy.org homepage. A subsequent PR will incorporate the Algolia interface for search and listing resources by category.

- [x] Replace original vanilla JS demo with Gatsby site
- [x] Configure ESLint pre-commit/prettier and other linters
- [x] Create the global styles and layout based on learn.astropy.org
   - [x] Header
   - [x] Search box
   - [x] Navigation cards
   - [x] Content
 - [x] Markdown based contribution documentation

### CODE REVIEW

**Comment by jonathansick** [comment]
ok! I've got the basic structure of the Gatsby site ready. I've also included a mechanism to add contribution documentation as Markdown (so it's easy to maintain). I've only copied the existing docs over; in a future PR we can update and expand the docs for contributors.

The next step is to add the search UI with Algolia.

`/`

localhost_8000 Laptop 2021-04-01 18 39 45

`/contributing` (a markdown page)

localhost_8000_contributing Laptop 2021-04-01 18 40 51

`/contributing/how-to-contribute` (a markdown documentation page)

localhost_8000_contributing_how-to-contribute Laptop 2021-04-01 18 44 12

File: README.md [hunk 1]
```diff
@@ -1,17 +1,14 @@
-# Astropy Search Demo
+# Astropy Learn
 
-This repository is a demonstration of a documentation search service for the [Astropy Project](https://www.astropy.org).
-The site itself is built around [Algolia](https://www.algolia.com).
-Records for the Algolia database are curated and formatted by the [astropy-librarian](https://github.com/jonathansick/astropy-librarian) app.
-Ultimately this search functionality will be built into the [learn.astropy.org](https://learn.astropy.org) site; in the meantime this stripped-down site lets us explore the interplay between front-end search experience and back-end data preparation.
+This repository the (future) homepage of the Astropy Learn project, https://learn.astropy.org. The site itself is built with [Gatsby](https://www.gatsbyjs.com/) and the [Algolia](https://www.algolia.com) search service. Records for the Algolia database are curated and formatted by the [astropy-librarian](https://github.com/jonathansick/astropy-librarian) app.
```
Comment (adrn) [inline README.md:3]:
```suggestion
This repository hosts the (future) homepage of the Astropy Learn project, https://learn.astropy.org. The site itself is built with [Gatsby](https://www.gatsbyjs.com/) and the [Algolia](https://www.algolia.com) search service. Records for the Algolia database are curated and formatted by the [astropy-librarian](https://github.com/jonathansick/astropy-librarian) app.
```

File: src/contributing/index.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+title: 'Contributing to Learn Astropy'
+slug: '/contributing/'
+---
+
+## Contributing content
+
+- [How to contribute tutorials to Learn Astropy](./contributing/how-to-contribute)
```
Comment (adrn) [inline src/contributing/index.md:8]:
```suggestion
- [How to contribute tutorials to Learn Astropy](/contributing/how-to-contribute)
```

File: src/contributing/index.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+## Learn Astropy infrastructure
+
+- [About the Tutorials Infrastructure](./contributing/infrastructure)
```
Comment (adrn) [inline src/contributing/index.md:12]:
```suggestion
- [About the Tutorials Infrastructure](/contributing/infrastructure)
```

File: src/styles/globalStyles.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  a {
+    color: var(--astropy-primary-color);
+    font-weight: 700;
+    text-decoration: none;
+  }
+
+  a:hover {
+    text-decoration: solid underline var(--astropy-primary-color) 2px;;
```
Comment (adrn) [inline src/styles/globalStyles.js:68]:
```suggestion
    text-decoration: solid underline var(--astropy-primary-color) 2px;
```

File: src/components/layout.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  /* Flexbox for the sticky footer */
+  display: flex;
+  flex-direction: column;
+  min-height: 100vh;
+
+  .upper-container {
+    flex: 1 0 auto;
+  }
+  .sticky-footer-container {
+    flex-shink: 0;
```
Comment (adrn) [inline src/components/layout.js:33]:
```suggestion
    flex-shrink: 0;
```

File: src/components/seo.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Helmet from 'react-helmet';
+import { useStaticQuery, graphql } from 'gatsby';
+
+/*
+ * SEO component that adds tags to teh page's header using react-helmet.
```
Comment (adrn) [inline src/components/seo.js:7]:
```suggestion
 * SEO component that adds tags to the page's header using react-helmet.
```

File: src/components/footer.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      </p>
+      <ul>
+        <li>
+          <Link to="/">Guides</Link>
+        </li>
+        <li>
+          <Link to="/">Tutorials</Link>
+        </li>
+        <li>
+          <Link to="/">Documention</Link>
```
Comment (adrn) [inline src/components/footer.js:28]:
```suggestion
          <Link to="/">Documentation</Link>
```

File: src/components/seo.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+            author
+            siteUrl
+            twitter
+          }
+        }
+      }
+    `
+  );
+
+  // The description can be overriden for individual pages via description
```
Comment (adrn) [inline src/components/seo.js:26]:
```suggestion
  // The description can be overidden for individual pages via description
```

**Review by adrn — COMMENTED** [review]
Looks good to me, aside from a few minor comments!

**Comment by jonathansick** [comment]
Thanks for the fixes @adrn !

**Pull request merged**